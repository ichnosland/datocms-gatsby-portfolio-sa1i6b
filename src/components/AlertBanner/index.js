/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
