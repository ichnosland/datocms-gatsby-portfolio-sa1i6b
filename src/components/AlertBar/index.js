/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
