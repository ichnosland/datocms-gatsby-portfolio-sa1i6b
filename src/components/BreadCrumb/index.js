/**
*
* BreadCrumb
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BreadCrumb = styled(Div)`
  color: ${(props) => props.color};
  position: relative;
  display: ${(props) => props.display ? props.display : 'inline-block'};
  text-align: center;
  border: ${(props) => props.border ? props.border : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const Crumb = styled(Div)`
  position: relative;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-wrap: nowrap;
  padding: 0.4em 1.4em 0.4em 0.8em;
  &:first-child {
    padding-left: 1.2em;
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:not(:first-child):not(:last-child) {
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:only-child {
    &:after {
      content: "";
      position: relative;
    }
  }
`;

BreadCrumb.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export default BreadCrumb;
