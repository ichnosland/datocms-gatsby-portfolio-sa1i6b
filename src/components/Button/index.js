/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
