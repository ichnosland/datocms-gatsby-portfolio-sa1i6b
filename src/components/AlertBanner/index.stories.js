import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select, color } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import 'global-styles';
import { allThemes, themeOptions } from 'style/theme';
import { actionColorOptions } from 'style/color';
import Div from '../Div';
import { Icon } from '../Button';
import buttonicon from 'icons/buttons';
import FlexBox from '../FlexBox';
import AlertBanner, { InfoBanner } from './index';

const icona = Object.keys(buttonicon).map((icn) => (icn));

const nota = `
  La props 'border' qui è boolean ma viene sovrascritta da actioncolor.
`;

storiesOf('AlertBanner', module)
  .addDecorator(withKnobs)
  .add('AlertBanner', withNotes(nota)(() => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <AlertBanner
        onClick={action('clicked')}
        actioncolor={select('actioncolor', actionColorOptions)}
        tag={boolean('tag', false)}
        color={color('color ')}
        bgcolor={color('bgcolor ')}
        margin={text('margin ')}
        padding={text('padding ')}
        border={boolean('border ', false)}
        shadow={boolean('shadow ', false)}
        radius={text('radius ')}
      >
        {boolean('con icona', false) &&
          <Icon {...buttonicon[select('icona', icona, 'cancel')]} left />
        }
        {text('Text', 'AlertBanner')}
      </AlertBanner>
    </ThemeProvider>
  )))
  .add('Alert ritiro', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <AlertBanner
        actioncolor="action"
        tag
      >
        <Icon {...buttonicon.hourglass} left />
        <span> &nbsp;Versione in fase di valutazione</span>
      </AlertBanner>
    </ThemeProvider>
  ))
  .add('InfoBanner', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <Div>
        <InfoBanner
          onClick={action('clicked')}
          actioncolor={select('actioncolor', actionColorOptions)}
          tag={boolean('tag', false)}
          color={color('color ')}
          bgcolor={color('bgcolor ')}
          margin={text('margin ')}
          padding={text('padding ')}
          border={boolean('border ', false)}
          radius={text('radius ')}
        >
          <span>Testo</span>
        </InfoBanner>
        <FlexBox maxWidth="480px" margin="60px auto 0" alignItems="stretch">
          <InfoBanner
            onClick={action('clicked')}
            actioncolor={select('actioncolor', actionColorOptions)}
            tag={boolean('tag', false)}
            color={color('color ')}
            bgcolor={color('bgcolor ')}
            margin={text('margin ')}
            padding={text('padding ')}
            border={boolean('border ', false)}
            radius={text('radius ')}
          >
            <span>Testo più lungo</span>
          </InfoBanner>
          <InfoBanner
            onClick={action('clicked')}
            actioncolor={select('actioncolor', actionColorOptions)}
            tag={boolean('tag', false)}
            color={color('color ')}
            bgcolor={color('bgcolor ')}
            margin={text('margin ')}
            padding={text('padding ')}
            border={boolean('border ', false)}
            radius={text('radius ')}
          >
            <span>Tempo stimato esecuzione: 30&apos;</span>
          </InfoBanner>
          <InfoBanner
            onClick={action('clicked')}
            actioncolor={select('actioncolor', actionColorOptions)}
            tag={boolean('tag', false)}
            color={color('color ')}
            bgcolor={color('bgcolor ')}
            margin={text('margin ')}
            padding={text('padding ')}
            border={boolean('border ', false)}
            radius={text('radius ')}
          >
            <span>Data assegnazione: 30/12/2018</span>
          </InfoBanner>
        </FlexBox>
      </Div>
    </ThemeProvider>
  ));
