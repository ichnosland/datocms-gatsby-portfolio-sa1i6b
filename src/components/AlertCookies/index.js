/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
