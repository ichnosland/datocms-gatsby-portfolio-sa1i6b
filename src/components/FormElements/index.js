/**
*
* FormElements
*
*/

import styled, { css } from 'styled-components';

import Section from 'components/Section';
import Div from 'components/Div';
import { Button, Icon } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const BasicInputStyle = css`
  display: block;
  box-sizing: border-box;
  color: ${(props) => props.theme.brand};
  width: 100%;
  text-align: ${(props) => props.align ? props.align : 'left'};
  appearance: none;
  padding: 6px 13px;
  border: 1px solid ${(props) => props.theme.light};
  background-color: ${colore.ui.contrast};
  ${({ note }) => note && css`
    color: ${colore.ui.txt};
    background-color: ${colore.ui.note};
    border-color: ${colore.ui.noteBorder};
  `}
  &::placeholder {
    position: relative;
    color: ${(props) => props.theme.subtle};
    ${({ note }) => note && css`
      color: ${colore.ui.noteBorder};
    `}
    opacity: 1;
    overflow: visible;
  }
  &:active,
  &:focus {
    outline: none;
  }
`;

BasicInputStyle.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

export const FormSection = styled(Section)`
  height: 100%;
  flex-direction: column;
  justify-content: space-between;
  &::before {
    content: "";
    display: inline;
  }
`;

export const CardForm = styled.form`
  width: 280px;
  margin: 0 auto;
  text-align: center;
  ${media.lt480`
    width: 240px;
  `}
`;

export const FormElement = styled(Div)`
  position: relative;
  margin: ${(props) => props.margin ? props.margin : '0 auto 18px'};
  ${(props) => props.errore && css`
    margin-bottom: 9px;
  }`
}`;


export const FormMessage = styled.span`
  display: inline-block;
  width: 100%;
  font-size: 15px;
  color: ${(props) => props.color ? props.color : colore.actions.error};
  padding: 6px 9px;
  margin-top: 9px;
  border: 1px solid ${colore.ui.neutralBg};
  background-color: ${colore.ui.contrast};
  border-radius: ${(props) => props.theme.radius.general};
  ${(props) => props.landing && css`
    font-size: smaller;
    padding: 2px 12px;
    border-radius: 60px;
    margin-bottom: 3px;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
`;

FormMessage.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export const FacebookButton = styled(Button)`
  background-color: ${colore.social.facebook};
  border-radius: 3px;
`;

export const FacebookIcon = styled(Icon)`
  top: -2px;
  fill: #fff;
`;
