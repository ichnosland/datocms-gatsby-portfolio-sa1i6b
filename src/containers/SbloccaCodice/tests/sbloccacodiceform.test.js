import React from 'react';
import { shallow } from 'enzyme';

import AlertBanner from 'components/AlertBanner';
import { CardForm } from 'components/FormElements';
import { Button } from 'components/Button';
import {
  FormTitle,
  SbloccaCodiceForm,
  required,
  maskDefinitions,
} from '../SbloccaCodiceForm';


const sbloccaCodiceFormProps = {
  feedback: {
    hasFeedback: false,
    tipologia: 'help',
    messaggio: 'messaggio',
  },
  handleSubmit: () => { },
  togglePaypalBox: () => { },
  valid: true,
};

describe('<SbloccaCodiceForm />', () => {
  it('visualizza un container con il form principale al suo interno', () => {
    const renderedComponent = shallow(
      <SbloccaCodiceForm
        {...sbloccaCodiceFormProps}
      />
    );
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent.find(FormTitle).length).toBe(2);
    expect(renderedComponent.find(AlertBanner).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('visualizza un container con un messaggio di feedback', () => {
    const props = {
      ...sbloccaCodiceFormProps,
      feedback: {
        ...sbloccaCodiceFormProps.feedback,
        hasFeedback: true,
      },
    };
    const renderedComponent = shallow(
      <SbloccaCodiceForm
        {...props}
      />
    );
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent.find(FormTitle).length).toBe(2);
    expect(renderedComponent.find(AlertBanner).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('CardForm > Button deve essere enabled quando valid = true', () => {
    const props = {
      ...sbloccaCodiceFormProps,
      valid: true,
    };

    const renderedComponent = shallow(
      <SbloccaCodiceForm
        {...props}
      />
    );

    expect(renderedComponent.find(Button).at(0).props().disabled).toBe(false);
  });

  it('CardForm > Button deve essere disabled quando valid = false', () => {
    const props = {
      ...sbloccaCodiceFormProps,
      valid: false,
    };

    const renderedComponent = shallow(
      <SbloccaCodiceForm
        {...props}
      />
    );

    expect(renderedComponent.find(Button).at(0).props().disabled).toBe(true);
  });
});

describe('<required />', () => {
  it('controllo che il comportamento della funzione required sia corretto', () => {
    expect(required('valore')).toBe(undefined);
    expect(required('')).toBe('Campo obbligatorio');
  });
});

describe('<maskDefinitions />', () => {
  it('controllo che il comportamento del maskDefinitions.A.transform sia corretto', () => {
    expect(maskDefinitions.A.transform('abcd-908-defg')).toBe('ABCD-908-DEFG');
  });
});
