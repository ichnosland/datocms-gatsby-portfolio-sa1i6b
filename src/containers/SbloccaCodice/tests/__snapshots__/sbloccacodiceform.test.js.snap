// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SbloccaCodiceForm /> visualizza un container con il form principale al suo interno 1`] = `
<FormElements__CardForm
  onSubmit={[Function]}
>
  <SbloccaCodiceForm__FormTitle
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    Se possiedi già un codice
  </SbloccaCodiceForm__FormTitle>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Inserisci qui il codice ricevuto dal tuo docente o quello che trovi sul frontespizio del libro.
  </Paragraph>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Assicurati di inserire tutti e 12 i caratteri da cui è composto.
  </Paragraph>
  <FormElements__FormElement>
    <Field
      align="center"
      autoComplete="off"
      component={[Function]}
      format={[Function]}
      label="codice"
      name="codice"
      normalize={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="text"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Button
      actioncolor="okay"
      disabled={false}
      full={true}
      theme={
        Object {
          "brand": "#00BBEF",
          "fonts": Object {
            "buttons": Object {
              "fontFamily": "inherit",
              "fontWeight": "inherit",
            },
          },
          "radius": Object {
            "buttons": "6px",
          },
        }
      }
    >
      Attiva Premium
    </Button>
  </FormElements__FormElement>
  <Paragraph__Separator
    theme={
      Object {
        "light": "#C9E4FF",
      }
    }
  >
    <Paragraph__SeparatorText>
      Oppure
    </Paragraph__SeparatorText>
  </Paragraph__Separator>
  <SbloccaCodiceForm__FormTitle
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    Procedi all'acquisto
  </SbloccaCodiceForm__FormTitle>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Assicurati di acquistare una licenza Premium solo se sei stato invitato a far parte di una classe dal tuo docente.
  </Paragraph>
  <FormElements__FormElement>
    <Button
      actioncolor=""
      full={true}
      onClick={[Function]}
      outline={1}
      theme={
        Object {
          "brand": "#00BBEF",
          "fonts": Object {
            "buttons": Object {
              "fontFamily": "inherit",
              "fontWeight": "inherit",
            },
          },
          "radius": Object {
            "buttons": "6px",
          },
        }
      }
      type="button"
    >
      Acquista Premium
    </Button>
  </FormElements__FormElement>
  <Image
    alt="PayPal"
    responsive={true}
    src="IMAGE_MOCK"
  />
  <Paragraph__LegalInfo>
    PayPal è il metodo rapido e sicuro per pagare e farsi pagare online
  </Paragraph__LegalInfo>
</FormElements__CardForm>
`;

exports[`<SbloccaCodiceForm /> visualizza un container con un messaggio di feedback 1`] = `
<FormElements__CardForm
  onSubmit={[Function]}
>
  <SbloccaCodiceForm__FormTitle
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    Se possiedi già un codice
  </SbloccaCodiceForm__FormTitle>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Inserisci qui il codice ricevuto dal tuo docente o quello che trovi sul frontespizio del libro.
  </Paragraph>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Assicurati di inserire tutti e 12 i caratteri da cui è composto.
  </Paragraph>
  <AlertBanner
    actioncolor="help"
    text="messaggio"
    theme={
      Object {
        "brand": "#00BBEF",
        "radius": Object {
          "general": "6px",
        },
      }
    }
  >
    messaggio
  </AlertBanner>
  <FormElements__FormElement>
    <Field
      align="center"
      autoComplete="off"
      component={[Function]}
      format={[Function]}
      label="codice"
      name="codice"
      normalize={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="text"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Button
      actioncolor="okay"
      disabled={false}
      full={true}
      theme={
        Object {
          "brand": "#00BBEF",
          "fonts": Object {
            "buttons": Object {
              "fontFamily": "inherit",
              "fontWeight": "inherit",
            },
          },
          "radius": Object {
            "buttons": "6px",
          },
        }
      }
    >
      Attiva Premium
    </Button>
  </FormElements__FormElement>
  <Paragraph__Separator
    theme={
      Object {
        "light": "#C9E4FF",
      }
    }
  >
    <Paragraph__SeparatorText>
      Oppure
    </Paragraph__SeparatorText>
  </Paragraph__Separator>
  <SbloccaCodiceForm__FormTitle
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    Procedi all'acquisto
  </SbloccaCodiceForm__FormTitle>
  <Paragraph
    theme={
      Object {
        "brand": "#00BBEF",
        "darken": "#1588AA",
      }
    }
  >
    Assicurati di acquistare una licenza Premium solo se sei stato invitato a far parte di una classe dal tuo docente.
  </Paragraph>
  <FormElements__FormElement>
    <Button
      actioncolor=""
      full={true}
      onClick={[Function]}
      outline={1}
      theme={
        Object {
          "brand": "#00BBEF",
          "fonts": Object {
            "buttons": Object {
              "fontFamily": "inherit",
              "fontWeight": "inherit",
            },
          },
          "radius": Object {
            "buttons": "6px",
          },
        }
      }
      type="button"
    >
      Acquista Premium
    </Button>
  </FormElements__FormElement>
  <Image
    alt="PayPal"
    responsive={true}
    src="IMAGE_MOCK"
  />
  <Paragraph__LegalInfo>
    PayPal è il metodo rapido e sicuro per pagare e farsi pagare online
  </Paragraph__LegalInfo>
</FormElements__CardForm>
`;
